File: Algorytmy i struktury danych/Python/AiSD5_C2 Ćwiczenie 2/zad3_suma_cyfr.py
from math import trunc
def Czytaj_liczbe():
    return int(input("Podaj liczbe: "))
def Suma_cyfr(liczba):
    suma=0
    liczba=abs(liczba)
    while(liczba>0):
        suma=suma+(liczba%10)
        liczba=int(trunc(liczba/10))
    return suma
def Wypisz_wynik(liczba,suma):
    print("Suma cyfr liczby",liczba,"to",suma)
def Glowna_procedura():
    liczba=Czytaj_liczbe()
    Wypisz_wynik(liczba,Suma_cyfr(liczba))
Glowna_procedura()

File: Algorytmy i struktury danych/Pascal/AiSD5_C2 Ćwiczenie 2/zad3_suma_cyfr.pas
program suma_cyfr;
function Czytaj_liczbe : Integer;
var
   liczba : Integer;
begin
   write('Podaj liczbe: ');
   readln(liczba);
   Czytaj_liczbe:=liczba;
end;

function Suma_cyfr(liczba : Integer) : Integer;
var suma : Integer;
begin
   suma:=0;
   liczba:=abs(liczba);
   while liczba>0 do
   begin
      suma:=suma+(liczba mod 10);
      liczba:=trunc(liczba/10);
   end;
   Suma_cyfr:=suma;
end;

procedure Wypisz_wynik(liczba : Integer; suma : Integer );
begin
   writeln('Suma cyfr liczby ', liczba, ' to ', suma);
end;

procedure Glowna_procedura;
var liczba : Integer;
begin
   liczba:=Czytaj_liczbe();
   Wypisz_wynik(liczba,Suma_cyfr(liczba));
end;

begin
   Glowna_procedura();
end.
